--- chkconfig-1.2.24h/Makefile.am.orig	Fri Apr 19 06:12:57 2002
+++ chkconfig-1.2.24h/Makefile.am	Fri May 14 10:57:16 2004
@@ -3,6 +3,7 @@
 sbin_PROGRAMS = chkconfig @NTSYSV@
 chkconfig_SOURCES = leveldb.c leveldb.h chkconfig.c
 ntsysv_SOURCES = leveldb.c leveldb.h ntsysv.c
+ntsysv_LDADD = -lnewt
 SUBDIRS = intl po
 man_MANS = chkconfig.8 @NTSYSVMAN@
 
--- chkconfig-1.2.24h/configure.in.orig	Fri May 14 10:53:42 2004
+++ chkconfig-1.2.24h/configure.in	Fri May 14 10:57:09 2004
@@ -23,7 +23,7 @@
 
 dnl Checks for libraries.
 if (test "$COMP_NTSYSV" = "yes"); then
-    AC_CHECK_LIB(newt, newtInit, , NTSYSV=""; NTSYSVMAN="")
+    AC_CHECK_LIB(newt, newtInit, [LIBNEWT=-lnewt], NTSYSV=""; NTSYSVMAN="")
 fi
 AC_CHECK_LIB(popt, poptReadDefaultConfig)
 AC_CHECK_LIB(getopt, getopt_long)
