--- chkconfig-1.30/Makefile.orig	2024-12-08 12:53:49.284711705 +0100
+++ chkconfig-1.30/Makefile	2024-12-08 12:54:37.537783629 +0100
@@ -2,8 +2,8 @@ VERSION := $(shell awk '/Version:/ { pri
 NEXT_VERSION := $(shell sed -nr 's/Version:[ ]*([0-9]*)\.([0-9]*)/echo "\1\.$$((\2+1))"/gep' chkconfig.spec)
 TAG = $(VERSION)
 
-CFLAGS = -g -Wall $(RPM_OPT_FLAGS) -D_GNU_SOURCE
-LDFLAGS += -g
+CFLAGS = $(OPTFLAGS) -Wall -D_GNU_SOURCE
+LDFLAGS += $(OPTLDFLAGS)
 MAN = chkconfig.8 ntsysv.8 alternatives.8
 PROG = chkconfig
 SBINDIR = /usr/sbin
