diff -Nur chkconfig-0.9.5/Makefile chkconfig-0.9.5.orig/Makefile
--- chkconfig-0.9.5/Makefile	Sun Sep 27 19:53:01 1998
+++ chkconfig-0.9.5.orig/Makefile	Wed Dec  9 11:06:06 1998
@@ -1,8 +1,8 @@
 VERSION=$(shell awk '/define version/ { print $$3 }' chkconfig.spec)
 CVSTAG = r$(subst .,-,$(VERSION))
-
-CFLAGS=-g -Wall $(RPM_OPT_FLAGS)
-LDFLAGS=-g
+OPTIMIZE = -g -O2
+CFLAGS= -Wall $(OPTIMIZE)
+LDFLAGS=-s
 MAN=chkconfig.8 ntsysv.8
 PROG=chkconfig
 BINDIR = /sbin
 
