diff -Nur chkconfig-1.2.16.orig/Makefile chkconfig-1.2.16/Makefile
--- chkconfig-1.2.16.orig/Makefile	Fri Mar 16 17:08:02 2001
+++ chkconfig-1.2.16/Makefile	Fri Mar 16 17:11:03 2001
@@ -22,7 +22,7 @@
 	done && test -z "$$fail"
 
 chkconfig: $(OBJS)
-	$(CC) $(LDFLAGS) -o chkconfig $(OBJS) /usr/lib/libpopt.a
+	$(CC) $(LDFLAGS) -o chkconfig $(OBJS) -lpopt
 
 ntsysv: $(NTOBJS)
 	$(CC) $(LDFLAGS) -o ntsysv $(NTOBJS) -lnewt -lpopt $(LIBMHACK)
