diff -Nru chkconfig-1.2.24h.old/chkconfig.c chkconfig-1.2.24h/chkconfig.c
--- chkconfig-1.2.24h.old/chkconfig.c	2003-12-03 22:48:54.000000000 +0100
+++ chkconfig-1.2.24h/chkconfig.c	2003-12-03 22:01:10.000000000 +0100
@@ -116,7 +116,7 @@
     printf("%-15s", s.name);
 
     for (i = 0; i <= conf.maxlevel; i++) {
-	printf("\t%d:%s", i, isOn(s.name, i) ? _("on") : _("off"));
+	printf("\t%d:%s", i, isOn(s.name, i) ? "***" : "---");
     }
     printf("\n");
 

