diff -urNpa chkconfig-1.15.orig/Makefile chkconfig-1.15/Makefile
--- chkconfig-1.15.orig/Makefile	2021-01-21 09:19:06.000000000 +0100
+++ chkconfig-1.15/Makefile	2021-05-04 10:44:37.807341418 +0200
@@ -2,8 +2,8 @@ VERSION := $(shell awk '/Version:/ { pri
 NEXT_VERSION := $(shell awk '/Version:/ { print $$2 + 0.01 }' chkconfig.spec)
 TAG = $(VERSION)
 
-CFLAGS = -g -Wall $(RPM_OPT_FLAGS) -D_GNU_SOURCE
-LDFLAGS += -g
+CFLAGS = $(OPTFLAGS) -Wall -D_GNU_SOURCE
+LDFLAGS += $(OPTLDFLAGS)
 MAN = chkconfig.8 ntsysv.8 alternatives.8
 PROG = chkconfig
 BINDIR = /sbin
