--- chkconfig-1.3.20/chkconfig.c~	2005-10-22 15:39:22.000000000 +0200
+++ chkconfig-1.3.20/chkconfig.c	2005-10-22 15:43:32.000000000 +0200
@@ -187,6 +187,7 @@
     int rc;
     int i;
     struct service s;
+	char *on, *off;
 
     rc = readServiceInfo(name, &s, 0);
     
@@ -207,13 +208,17 @@
     }
 
     printf("%-15s", s.name);
+
+	on = getenv("CHKCONFIG_ON");
+	off = getenv("CHKCONFIG_OFF");
+	
     if (s.type == TYPE_XINETD) {
-	    printf("\t%s\n", s.levels ? _("on") : _("off"));
+	    printf("\t%s\n", s.levels ? on ? on : _("on") : off ? off : _("off"));
 	    return 0;
     }
 
     for (i = 0; i < 7; i++) {
-	printf("\t%d:%s", i, isOn(s.name, i) ? _("on") : _("off"));
+		printf("\t%d:%s", i, isOn(s.name, i) ? on ? on : _("on") : off ? off : _("off"));
     }
     printf("\n");
 
