--- chkconfig-1.3.58/chkconfig.c~	2012-12-04 15:41:14.000000000 +0200
+++ chkconfig-1.3.58/chkconfig.c	2012-12-04 15:42:06.080488828 +0200
@@ -27,6 +27,7 @@
 #include <unistd.h>
 
 static char *progname;
+extern int addItem;
 
 #define _(String) gettext((String)) 
 
@@ -243,12 +244,10 @@
 		frobDependencies(&s);
     else
     for (i = 0; i < 7; i++) {
-	if (!isConfigured(name, i, NULL, NULL)) {
 	    if ((1 << i) & s.levels)
 		doSetService(s, i, 1);
 	    else
 		doSetService(s, i, 0);
-	}
     }
 
     return 0;
@@ -626,7 +626,7 @@
 }
 
 int main(int argc, const char ** argv) {
-    int listItem = 0, addItem = 0, delItem = 0, overrideItem = 0, noRedirectItem = 0;
+    int listItem = 0, delItem = 0, overrideItem = 0, noRedirectItem = 0;
     int type = TYPE_ANY;
     int rc, i, x;
     char * levels = NULL;
diff -urN chkconfig-1.3.37.org/leveldb.c chkconfig-1.3.37/leveldb.c
--- chkconfig-1.3.37.org/leveldb.c	2007-08-03 19:57:56.000000000 +0200
+++ chkconfig-1.3.37/leveldb.c	2008-04-24 09:27:34.124665839 +0200
@@ -26,6 +26,10 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#include <sys/types.h>
+#include <dirent.h>
+
+int addItem = 0;
 
 /* Changes
    1998-09-22 - Arnaldo Carvalho de Melo <acme@conectiva.com.br>
@@ -563,6 +567,25 @@
 	    serv.provides[1] = NULL;
     }
 
+    if (addItem) {
+	    DIR *dir;
+	    struct dirent *dirent;
+	    int i;
+
+	    strcpy(levelbuf,"/etc/rc.d/rc0.d");
+	    for(i=0;i<7;i++,levelbuf[12]++) {
+		    dir=opendir(levelbuf);
+		    if (!dir) return 1;
+		    while ((dirent=readdir(dir))) {
+			    if (strlen(dirent->d_name)<4) continue;
+			    if (strcmp(dirent->d_name+3,name)) continue;
+			    if (*dirent->d_name=='K') serv.levels &= ~(1<<i);
+			    else if (*dirent->d_name=='S') serv.levels |= 1<<i;
+		    }
+		    closedir(dir);
+	    }
+    }
+
     *service = serv;
     return 0;
 }
